src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -182,10 +193,10 @@ protected static QueryParseContext createContext() {
         return new QueryParseContext(index, queryParserService);
     }
 
-    private static void assertQueryHeader(XContentParser parser, String expectedParserName) throws IOException {
+    private void assertQueryHeader(XContentParser parser) throws IOException {
         assertThat(parser.nextToken(), is(XContentParser.Token.START_OBJECT));
         assertThat(parser.nextToken(), is(XContentParser.Token.FIELD_NAME));
-        assertThat(parser.currentName(), is(expectedParserName));
+        assertThat(parser.currentName(), is(testQuery.parserName()));
```

Comment:
Hmm, right your are talking about merging the parser/builder classes at some point? But I thought that regardless all queries will keep their name field as a kind of ID, maybe we call it queryName() then? Not sure about this, I can revert this for now, was only a minor detail.